tests/Foundation/Http/Middleware/TrimStringsTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Illuminate\Tests\Foundation\Http\Middleware;
+
+use Illuminate\Http\Request;
+use PHPUnit\Framework\TestCase;
+use Illuminate\Foundation\Http\Middleware\TrimStrings;
+
+class TrimStringsTest extends TestCase
+{
+    public function testTrimSimpleFields()
+    {
+        $middleware = new TrimStrings;
+        $request = new Request(
+            [
+                'field1' => 'value1',
+                'field2' => '  value2  ',
+            ]
+        );
+
+        $middleware->handle($request, function (Request $request) {
+            $this->assertEquals('value1', $request->get('field1'));
+            $this->assertEquals('value2', $request->get('field2'));
+        });
+    }
+
+    /** @test */
+    public function testTrimArrayFields()
+    {
+        $middleware = new TrimStrings;
+        $request = new Request(
+            [
+                'field1' => ['value1', 'value2'],
+                'field2' => ['  value1  ', '  value2  '],
+            ]
+        );
+
+        $middleware->handle($request, function (Request $request) {
+            $t1 = $request->get('field1');
+            $this->assertEquals('value1', $t1[0]);
+            $this->assertEquals('value2', $t1[1]);
+
+            $t2 = $request->get('field2');
+            $this->assertEquals('value1', $t2[0]);
+            $this->assertEquals('value2', $t2[1]);
+        });
+    }
+
+    public function testTrimSimpleFieldsExceptSome()
+    {
+        $middleware = new TrimStringsExcept;
+
+        $request = new Request(
+            [
+                'field1' => 'value1',
+                'field2' => ' value2 ',
+            ]
+        );
+
+        $middleware->handle($request, function (Request $request) {
+            $this->assertEquals('value1', $request->get('field1'));
+            $this->assertEquals(' value2 ', $request->get('field2'));
+        });
+    }
+
+    public function testTrimArrayFieldsExceptSome()
+    {
+        $middleware = new TrimStringsExcept;
+        $request = new Request(
+            [
+                'field1' => ['value1', 'value2'],
+                'field2' => ['  value1  ', '  value2  '],
+            ]
+        );
+
+        $middleware->handle($request, function (Request $request) {
+            $t1 = $request->get('field1');
+            $this->assertEquals('value1', $t1[0]);
+            $this->assertEquals('value2', $t1[1]);
+
+            $t2 = $request->get('field2');
+            $this->assertEquals('  value1  ', $t2[0]);
+            $this->assertEquals('  value2  ', $t2[1]);
+        });
+    }
+}
```

Comment:
What would be the result of running the same middleware running on:

```html
<input name="nested[][field1]" value=" trimmed ">
<input name="nested[][field2]" value=" not trimmed ">
```

This is the current behaviour:

```php
$middleware = new class extends Illuminate\Foundation\Http\Middleware\TrimStrings {
    protected $except = ['field2'];
};

$request = new Illuminate\Http\Request([
    'nested' => [
        [
            'field1' => ' trimmed ',
            'field2' => ' not trimmed ',
        ],
    ],
]);

$middleware->handle($request, function ($request) {
    dump($request->input('nested'));
});
/*
[
  [
    "field1" => "trimmed"
    "field2" => " not trimmed "
  ]
]
*/
```

From what I can see, this pull request will change that behavior.

e.g., a registration form for many users may have multiple parameters `$request->input('user.*.password')` where `$except = ['password']`